Sources/Brave/WebFilters/BraveS3Resource.swift

```diff
@@ -0,0 +1,77 @@
+// Copyright 2023 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import Foundation
+import Shared
+
+enum BraveS3Resource: Hashable, DownloadResourceInterface {
```

Comment:
Maybe but then I would have to add custom filter lists here and make changes to the resource file anyways because I don't want to pass the security key to non s3 resources. I felt this was was more secure in the long term when we have a clear separation of what is going to s3 and what is not.